packages/platforms/react-native/tests/span-factory.test.ts

```diff
@@ -0,0 +1,196 @@
+
+    it('runs onSpanEnd callbacks for native spans', async () => {
+      spanFactory.onAttach()
+
+      const onSpanEndCallback = jest.fn((span) => {
+        return Promise.resolve(span.name === 'should send')
+      })
+
+      spanFactory.configure({ logger: jestLogger, onSpanEnd: [onSpanEndCallback] } as unknown as InternalConfiguration<ReactNativeConfiguration>)
+      const startTime = clock.now()
+      const validSpan = spanFactory.startSpan('should send', { startTime, isFirstClass: true })
+      expect(NativeBugsnagPerformance!.startNativeSpan).toHaveBeenCalledWith('should send', expect.objectContaining({ startTime: clock.toUnixNanoseconds(startTime) }))
+      expect(contextStorage.current).toBe(validSpan)
+
+      const invalidSpan = spanFactory.startSpan('should discard', { startTime, isFirstClass: true })
+      expect(NativeBugsnagPerformance!.startNativeSpan).toHaveBeenCalledWith('should discard', expect.objectContaining({ startTime: clock.toUnixNanoseconds(startTime) }))
+      expect(contextStorage.current).toBe(invalidSpan)
+
+      const endTime = clock.now()
+      spanFactory.endSpan(invalidSpan, endTime)
+      await jest.runOnlyPendingTimersAsync()
+
+      expect(onSpanEndCallback).toHaveBeenCalledTimes(1)
+      expect(NativeBugsnagPerformance!.endNativeSpan).not.toHaveBeenCalled()
+      expect(NativeBugsnagPerformance!.discardNativeSpan).toHaveBeenCalledTimes(1)
+
+      spanFactory.endSpan(validSpan, endTime)
+      await jest.runOnlyPendingTimersAsync()
+
+      expect(onSpanEndCallback).toHaveBeenCalledTimes(2)
+      expect(NativeBugsnagPerformance!.endNativeSpan).toHaveBeenCalledTimes(1)
+      expect(NativeBugsnagPerformance!.discardNativeSpan).toHaveBeenCalledTimes(1)
+
+      expect(processor.spans.length).toBe(0)
+    })
+  })
+
+  describe('discardSpan', () => {
+    it('discards a native span', () => {
+      spanFactory.onAttach()
+
+      const startTime = clock.now()
+      const nativeSpan = spanFactory.startSpan('native span', { startTime, isFirstClass: true })
+      expect(NativeBugsnagPerformance!.startNativeSpan).toHaveBeenCalledWith('native span', expect.objectContaining({ startTime: clock.toUnixNanoseconds(startTime) }))
+
+      backgroundingListener.sendToBackground()
+
+      spanFactory.endSpan(nativeSpan, clock.now())
+      expect(NativeBugsnagPerformance!.endNativeSpan).not.toHaveBeenCalled()
+      expect(NativeBugsnagPerformance!.discardNativeSpan).toHaveBeenCalledWith(nativeSpan.id, nativeSpan.traceId)
+    })
+
+    it('does not discard a non-native span', () => {
+      const startTime = clock.now()
+      const span = spanFactory.startSpan('not native', { startTime, isFirstClass: false })
+      expect(NativeBugsnagPerformance!.startNativeSpan).not.toHaveBeenCalled()
+
+      spanFactory.endSpan(span, clock.now())
+      expect(NativeBugsnagPerformance!.discardNativeSpan).not.toHaveBeenCalled()
+      expect(NativeBugsnagPerformance!.endNativeSpan).not.toHaveBeenCalled()
+    })
```

Comment:
what's calling `discardSpan` in this test? I assume it's the backgrounding in the previous test

Reply:
yeah, that's a mistake - good spot!

Reply:
I'll update both tests to use `DISCARD_END_TIME` to make things clearer